src/test/ui/const-eval/infinite_loop.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(const_let)]
+
```

Comment:
Just out of curiosity, how long does this test run before finishing with an error?

Reply:
I believe a second or two, I'll rerun it to make sure.

Reply:
> finished in 1.492

Reply:
For now I think it's fine. We should probably run a perf run first though

Reply:
If it is a problem, we can try some tricks like exponential backoff paired with clearing the HashSet